src/main/java/games/strategy/util/Match.java

```diff
@@ -96,70 +89,45 @@ private Match(final Predicate<T> condition) {
 
   /**
    * returns true if all elements in the collection match.
+   * returns false if the collection is empty.
+   */
+  public static <T> boolean allMatchNotEmpty(final Collection<T> collection, final Match<T> match) {
```

Comment:
Implementing collections also enables us to get rid of the `addAll(UnitCollection)` overload